Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -3752,16 +3752,22 @@ static String commandNameForSelector(SEL selector)
     return accessibilityFocusedUIElement();
 }
 
-void WebViewImpl::enableAccessibilityIfNecessary()
+void WebViewImpl::enableAccessibilityIfNecessary(NSString *attribute)
 {
-    if (WebCore::AXObjectCache::accessibilityEnabled())
-        return;
-
 #if ENABLE(INITIALIZE_ACCESSIBILITY_ON_DEMAND)
-    m_page->initializeAccessibility();
-    accessibilityRegisterUIProcessTokens();
+    // The attributes NSAccessibilityParentAttribute and NSAccessibilityPositionAttribute do not require AX initialization in the WebContent process.
+    if (!m_page->protectedLegacyMainFrameProcess()->processPool().hasReceivedAXRequestInUIProcess()
```

Comment:
We should cache `m_page->protectedLegacyMainFrameProcess()->processPool()` in a local. It is used twice.